components/hab/src/command/pkg/binlink.rs

```diff
@@ -208,16 +197,65 @@ impl Binlink {
         Err(Error::CannotParseBinlinkSource(path.as_ref().to_path_buf()))
     }
 
-    pub fn link(&self) -> Result<()> {
-        let template = format!("@echo off\nREM source='{0}'\n\"{0}\" %*",
-                               self.src.display());
-        fs::write(&self.dest, template)?;
+    pub fn link(&self, env: HashMap<String, String>) -> Result<()> {
+        self.write_binstub(self.stub_template(env).as_bytes())
+    }
+
+    #[cfg(not(target_os = "windows"))]
+    fn stub_template(&self, env: HashMap<String, String>) -> String {
+        let mut exports = String::new();
+        for (key, value) in env.into_iter() {
+            exports.push_str(&format!("export {}={}\n", key, value));
+        }
+        format!(
+                "\
+#!/bin/sh
+# source='{0}'
+{1}
+{0} \"$@\"",
+                self.src.display(),
+                exports
+        )
+    }
+
+    #[cfg(target_os = "windows")]
+    fn stub_template(&self, env: HashMap<String, String>) -> String {
+        let mut exports = String::new();
+        for (key, value) in env.into_iter() {
+            exports.push_str(&format!("SET {}={}\n", key, value));
+        }
+        format!(
+                "\
```

Comment:
If you'd like to be able to indent this multi-line string, I finally found a solution that works for me in the [indoc](https://crates.io/crates/indoc) crate which provides an `indoc!` macro to help with this problem. Here's an [example](https://github.com/fnichol/mtoc/blob/aebbfe6ada535dcb8eede38f3f2084c0760f31fb/mtoc-parser/src/write.rs#L1031-L1047) with a raw string to boot.